internal/xds/translator/listener.go

```diff
@@ -92,6 +93,21 @@ func addXdsHTTPFilterChain(xdsListener *listener.Listener, irListener *ir.HTTPLi
 		}},
 	}
 
+	// Enable grpc-web filter for HTTP2
+	if irListener.IsHTTP2 {
```

Comment:
from the godoc on `IsHTTP2`:

```
// IsHTTP2 is set if the upstream client as well as the downstream server are configured to serve HTTP2 traffic.
```

this is a little misleading i think? since the codec type is always set to "auto" and will infer the downstream protocol which could be HTTP/1 and upgraded when talking with the backend to gRPC now that this filter is applied

Reply:
> 

How should I configure it so that it can touch it ? @sunjayBhatia

Reply:
good point @sunjayBhatia, we could limit `IsHTTP2` to downstreams speaking HTTP/2, thereby limiting `grpc-web` for HTTP/2 only clients

Reply:
updated the PR with the codec type change from auto to http2

Reply:
seems useful for clients to be able to use gRPC-Web over HTTP/1 -> Envoy -> gRPC backend over HTTP/2 as well! but seems like that could be a separate feature if desired